File: handson/dotnet/Util/TokenUtil.cs
namespace AwesomePizza.Util;

public static class TokenUtil
{
    public static string? ExtractBearerToken(HttpRequest request)
    {
        var authHeader = request.Headers.Authorization.ToString();
        if (string.IsNullOrEmpty(authHeader) || !authHeader.StartsWith("Bearer "))
        {
            return null;
        }
        return authHeader[7..];
    }
}

File: handson/java/src/main/java/com/awesomepizza/util/TokenUtil.java
package com.awesomepizza.util;

import jakarta.servlet.http.HttpServletRequest;

public final class TokenUtil {

    private TokenUtil() {
    }

    public static String extractBearerToken(HttpServletRequest request) {
        String authHeader = request.getHeader("Authorization");
        if (authHeader == null || !authHeader.startsWith("Bearer ")) {
            return null;
        }
        return authHeader.substring(7);
    }
}
